card/mod/machines/lib/javascript/decko/filter.js.coffee

```diff
@@ -1,180 +1,99 @@
+#decko.slotReady (slot) ->
+#  slot.find("._filter-widget").each ->
+#    showDefaultFilters $(this)
+
 $.extend decko,
```

Comment:
two main changes in this file:
1. moved all the filtered list (filter items) stuff to a separate file.
2. restructured the filter handling to support resets.

The reset change involves starting with a full list of clean prototypes. Before there were only prototypes of hidden filters.  Now there's a full list, and we clone and initialize and show the active ones when the page is loaded.